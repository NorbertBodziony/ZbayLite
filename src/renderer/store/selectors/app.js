import { createSelector } from 'reselect'

const store = s => s

export const app = createSelector(store, state => state.get('app'))

const version = createSelector(app, a => a.version)

export default {
  version
}
