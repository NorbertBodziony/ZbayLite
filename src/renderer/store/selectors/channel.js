import { createSelector } from 'reselect'
import channelsSelector from './channels'
import identitySelectors from './identity'
import pendingMessagesSelectors from './pendingMessages'

const store = s => s

const channel = createSelector(store, state => state.get('channel'))

const data = createSelector(
  channelsSelector.data,
  channel,
  (channels, channel) => channels.find(ch => ch.get('id') === channel.id)
)

export const spentFilterValue = createSelector(channel, c => c.get('spentFilterValue', -1))

export const message = createSelector(channel, c => c.get('message'))

export const pendingMessages = createSelector(
  pendingMessagesSelectors.pendingMessages,
  channel,
  (pendingMessages, channel) => pendingMessages.filter(m => m.channelId === channel.id)
)

export const messagesMeta = createSelector(channel, c => c.messages)

const _isOwner = (address, message) => message.getIn(['sender', 'replyTo']) === address

export const messages = createSelector(
  identitySelectors.data,
  messagesMeta,
  pendingMessages,
  (identity, meta, pendingMessages) => {
    const displayableBroadcasted = meta.data.map(
      m => m.set('fromYou', _isOwner(identity.address, m))
    )
    const displayablePending = pendingMessages.map(
      m => m.get('message')
        .set('error', m.get('error'))
        .set('status', m.get('status'))
        .set('id', m.get('opId'))
        .set('fromYou', _isOwner(identity.address, m.get('message')))
    )
    return displayableBroadcasted.concat(displayablePending.values())
      .sortBy(m => m.get('createdAt'))
  })

export default {
  data,
  channel,
  spentFilterValue,
  message,
  pendingMessages,
  messages
}
