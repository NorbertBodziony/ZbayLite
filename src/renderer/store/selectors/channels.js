import { createSelector } from 'reselect'

const store = s => s

const channels = createSelector(store, state => state.get('channels'))
const data = createSelector(channels, ch => ch.data)
const loading = createSelector(channels, ch => ch.loading)
const errors = createSelector(channels, c => c.get('errors'))

export default {
  channels,
  loading,
  data,
  errors
}
