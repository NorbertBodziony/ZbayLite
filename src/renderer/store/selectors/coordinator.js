import { createSelector } from 'reselect'

const store = s => s

export const coordinator = createSelector(store, state => state.get('coordinator'))

const running = createSelector(coordinator, a => a.running)

export default {
  running
}
