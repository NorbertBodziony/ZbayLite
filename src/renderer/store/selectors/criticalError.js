import { createSelector } from 'reselect'

const store = s => s

const criticalError = createSelector(store, state => state.get('criticalError'))

const message = createSelector(criticalError, error => error.message)
const traceback = createSelector(criticalError, error => error.traceback)

export default {
  criticalError,
  message,
  traceback
}
