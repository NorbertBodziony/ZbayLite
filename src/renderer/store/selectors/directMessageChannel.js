import { createSelector } from 'reselect'

const store = s => s

export const directMessageChannel = createSelector(store, state => state.get('directMessageChannel'))

export default {
  directMessageChannel
}
