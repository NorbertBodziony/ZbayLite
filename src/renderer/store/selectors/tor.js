import { createSelector } from 'reselect'
const store = s => s

const tor = createSelector(
  store,
  state => state.get('tor')
)
const torEnabled = createSelector(
  tor,
  tor => tor.enabled
)

export default {
  tor,
  torEnabled
}
