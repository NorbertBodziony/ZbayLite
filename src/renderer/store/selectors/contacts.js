import { createSelector } from 'reselect'
import Immutable from 'immutable'
import identitySelectors from './identity'
import directMssagesQueueSelectors from './directMessagesQueue'
import zbayMessages from '../../zbay/messages'
import operationsSelectors from './operations'
import { operationTypes } from '../handlers/operations'
import usersSelectors from './users'
import { mergeIntoOne } from './channel'

export const Contact = Immutable.Record({
  lastSeen: null,
  key: '',
  username: '',
  address: '',
  messages: Immutable.Map(),
  newMessages: Immutable.List(),
  vaultMessages: Immutable.List(),
  offerId: null
})

const store = s => s

const contacts = createSelector(store, state => state.get('contacts'))
const contactsList = createSelector(contacts, identitySelectors.removedChannels, (contacts, removedChannels) => {
  if (removedChannels.size > 0) {
    return contacts.filter(c => c.key.length === 66 && c.offerId === null && !removedChannels.includes(c.address)).toList()
  }
  return contacts.filter(c => c.key.length === 66 && c.offerId === null).toList()
}
)

const offerList = createSelector(contacts, contacts =>
  contacts.filter(c => !!c.offerId).toList()
)
const channelsList = createSelector((contacts, identitySelectors.removedChannels), (contacts, removedChannels) => {
  if (removedChannels.size > 0) {
    return contacts.filter(c => c.key.length === 78 && c.offerId === null && !removedChannels.includes(c.address)).toList()
  }
  return contacts.filter(c => c.key.length === 78 && c.offerId === null).toList()
}
)

const directMessagesContact = address =>
  createSelector(contacts, c => c.toList().find(el => el.get('address') === address))

const contact = address =>
  createSelector(contacts, c => c.get(address, Contact()))
const messages = address =>
  createSelector(contact(address), c => c.messages.toList())
const allMessages = createSelector(contacts, c =>
  c.reduce((acc, t) => acc.merge(t.messages), Immutable.Map())
)
const getAdvertById = txid =>
  createSelector(allMessages, msgs => msgs.get(txid))
const lastSeen = address => createSelector(contact(address), c => c.lastSeen)
const username = address => createSelector(contact(address), c => c.username)
const vaultMessages = address =>
  createSelector(contact(address), c => c.vaultMessages)
const newMessages = address =>
  createSelector(contact(address), c => c.newMessages)

export const queuedMessages = address =>
  createSelector(
    directMssagesQueueSelectors.queue,
    queue =>
      queue.filter(
        m => m.recipientAddress === address && m.message.get('type') < 10
      ) //  separate offer messages and direct messages
  )

export const pendingMessages = address =>
  createSelector(operationsSelectors.operations, operations =>
    operations.filter(
      o =>
        o.type === operationTypes.pendingDirectMessage &&
        o.meta.recipientAddress === address &&
        o.meta.message.get('type') < 10 //  separate offer messages and direct messages
    )
  )

export const directMessages = (address, signerPubKey) =>
  createSelector(
    identitySelectors.data,
    usersSelectors.registeredUser(signerPubKey),
    messages(address),
    vaultMessages(address),
    (identity, registeredUser, messages, vaultMessages) => {
      const userData = registeredUser ? registeredUser.toJS() : null
      const identityAddress = identity.address
      const identityName = userData ? userData.nickname : identity.name

      const fetchedMessagesToDisplay = messages.map(msg =>
        zbayMessages.receivedToDisplayableMessage({
          message: msg,
          identityAddress,
          receiver: { replyTo: identityAddress, username: identityName }
        })
      )

      const concatedMessages = fetchedMessagesToDisplay
        .concat(vaultMessages.values())
        .sortBy(m => m.get('createdAt'))
      const merged = mergeIntoOne(concatedMessages)
      return merged
    }
  )

export default {
  contacts,
  directMessagesContact,
  queuedMessages,
  pendingMessages,
  contact,
  messages,
  directMessages,
  lastSeen,
  vaultMessages,
  username,
  newMessages,
  contactsList,
  channelsList,
  offerList,
  getAdvertById,
  allMessages
}
