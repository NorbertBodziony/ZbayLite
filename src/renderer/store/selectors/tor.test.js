/* eslint import/first: 0 */
jest.mock('../../vault')
import Immutable from 'immutable'

import create from '../create'
import selectors from './tor'
import { initialState } from '../handlers/tor'
describe('rates selectors', () => {
  let store = null
  beforeEach(() => {
    store = create({
      initialState: Immutable.Map({
        tor: initialState
      })
    })
    jest.clearAllMocks()
  })

  it('selects tor data', async () => {
    expect(selectors.tor(store.getState())).toMatchSnapshot()
  })
})
