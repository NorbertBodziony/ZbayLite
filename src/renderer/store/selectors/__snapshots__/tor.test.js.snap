// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rates selectors selects tor data 1`] = `
Immutable.Tor {
  "url": "",
  "enabled": false,
  "error": null,
  "status": "",
}
`;

exports[`rates selectors selects torEnabled 1`] = `false`;
